fhir-server/src/main/java/org/linuxforhealth/fhir/server/filter/rest/FHIRRestServletFilter.java

```diff
@@ -523,8 +530,38 @@ public void init(FilterConfig filterConfig) throws ServletException {
             shardKeyHeaderName = config.getStringProperty(FHIRConfiguration.PROPERTY_SHARD_KEY_HEADER_NAME,
                 FHIRConfiguration.DEFAULT_SHARD_KEY_HEADER_NAME);
             log.info("Configured shard-key header name is: '" +  shardKeyHeaderName + "'");
+            
+            if (filterConfig.getServletContext().getAttribute(FHIRServletContextListener.FHIR_CONFIGURED_TENANTS) != null 
+                    && filterConfig.getServletContext().getAttribute(FHIRServletContextListener.FHIR_CONFIGURED_TENANTS) instanceof Map<?, ?>) {
+                configuredTenants = (Map<String, String>) filterConfig.getServletContext().getAttribute(FHIRServletContextListener.FHIR_CONFIGURED_TENANTS);
+            }
         } catch (Exception e) {
             throw new ServletException("Servlet filter initialization error.", e);
         }
     }
+    
+    /**
+     * Checks if the tenant id is present in the list configured tenants. 
+     * If the tenant Id is present this method returns the value from the configured tenants list.
+     * @param tenantId the tenant for which the configured Tenants is going to be checked
+     * @return tenantId If the tenant Id is present this method returns the value from the configured tenants list.
+     * @throws Exception If the tenant Id is not present in the list configured tenants.
+     */
+    private String fetchValidConfiguredTenantId(String tenantId) throws Exception {
+        try {
+            if (configuredTenants != null && !configuredTenants.isEmpty() && configuredTenants.containsKey(tenantId)) {
```

Comment:
Doing 2 lookups into the Map (containsKey and get). Just do a get and check that the result is not null (because we never store null values in this map).

Reply:
This has been fixed as suggested. Removed the additional lookup from Map.